meilisearch/index.py

```diff
@@ -31,6 +31,7 @@ def __init__(
         primary_key: Optional[str] = None,
         created_at: Optional[Union[datetime, str]] = None,
         updated_at: Optional[Union[datetime, str]] = None,
+        http: Optional[HttpRequests] = None,
```

Comment:
Since `HttpRequests` comes from a private method it's not great to pass it around like this. I think it would be better to pass around the session because of this.

session also needs to be added to the docstring

```suggestion
        session: Optional[Session] = None,
```